src/utils/token-metadata.utils.ts

```diff
@@ -16,22 +17,45 @@ export interface TokenMetadataResponse {
   artifactUri?: string;
 }
 
+const transformDataToTokenMetadata = (
+  token: TokenMetadataResponse | TokenMetadataInterface | null,
+  address: string,
+  id: number
+) =>
+  !isDefined(token)
+    ? null
+    : {
+        id,
+        address,
+        decimals: token.decimals,
+        symbol: token.symbol ?? token.name?.substring(0, 8) ?? '???',
+        name: token.name ?? token.symbol ?? 'Unknown Token',
+        thumbnailUri: token.thumbnailUri,
+        artifactUri: token.artifactUri
+      };
+
 export const loadTokenMetadata$ = memoize(
   (address: string, id = 0): Observable<TokenMetadataInterface> =>
     from(tokenMetadataApi.get<TokenMetadataResponse>(`/metadata/${address}/${id}`)).pipe(
-      map(({ data }) => ({
-        id,
-        address,
-        decimals: data.decimals,
-        symbol: data.symbol ?? data.name?.substring(0, 8) ?? '???',
-        name: data.name ?? data.symbol ?? 'Unknown Token',
-        thumbnailUri: data.thumbnailUri,
-        artifactUri: data.artifactUri
-      }))
+      map(({ data }) => transformDataToTokenMetadata(data, address, id)),
+      filter(isDefined)
     ),
   { cacheKey: ([address, id]) => getTokenSlug({ address, id }) }
 );
 
+export const loadTokensMetadata$ = memoize(
+  (slugs: Array<string>): Observable<Array<TokenMetadataInterface>> =>
+    from(tokenMetadataApi.post<Array<TokenMetadataInterface | null>>('/', slugs)).pipe(
```

Comment:
mb TokenMetadataResponse should be here.  
if so, change transformDataToTokenMetadata